### PR TITLE
Remove memory_map sources and references

### PR DESCRIPTION
**Pull request by mikisch81**
- Remove _memory_map.h/c_.
- Remove includes for _memory_map.h_.
- Update _QUICKSTART.md_ with correct fault trace.

### CODE REVIEW

File: docs/lib/QUICKSTART.md [hunk 1]
```diff
@@ -451,17 +451,18 @@ Once the uVisor debug messages are enabled, your application fails. The failure
...
+* Active Box ID: 0
+* FAULT SYNDROME REGISTERS
 
-/* MEMORY MAP
-  Address:           0x4004800C
-  Region/Peripheral: SIM
-    Base address:    0x40047000
-    End address:     0x40048060
+  CFSR: 0x00008200
+  BFAR: 0x40048044
```
Comment (Patater) [inline docs/lib/QUICKSTART.md]: Where does this address come from? Does it make sense in the context of this getting started guide?

File: docs/lib/QUICKSTART.md [hunk 2]
```diff
@@ -451,17 +451,18 @@ Once the uVisor debug messages are enabled, your application fails. The failure
...
+  --> PRECISERR: precise data access.
 
 ...
 ```
 
+Look up the faulty address (The value of BFAR) in the target device reference manual.
```
Comment (Patater) [inline docs/lib/QUICKSTART.md]:
```diff
- The value of BFAR
+ the value of BFAR
```

**Comment by Patater** [comment]
Please explain why the memory map is being removed in the commit message.

File: docs/lib/QUICKSTART.md (hunk 1 above)
Comment (mikisch81) [inline docs/lib/QUICKSTART.md]: Yes, it was a part that showed a flow of a bus/memmanage fault due to missing ACL.
The other option is to remove entirely this part.

Comment (Patater) [inline docs/lib/QUICKSTART.md]: OK. I just want to make sure that if someone is following along with the guide, they see what we wrote here.

Comment (mikisch81) [inline docs/lib/QUICKSTART.md]: Hopefully soon the doc will be updated with the debug box traces.. :-)

**Pull request merged**